### PR TITLE
Inter list

### PR DESCRIPTION
**Pull request by rotshidzw**
- Add a new JavaScript file and import it as a module:
- it will contain methods related to the status updates (completed: true / false).
- Add event listener to the checkbox (change).
- Update items object's value for completed key upon user actions.
- Implement a function for the "Clear all completed" button (use filter() method).
- Store the updated array of items in local storage, so the user gets the correct list values after the page reloads.

### CODE REVIEW

File: modules/src/index.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    if (item.isCompleted) {
+      taskElement.classList.add('completed');
+    }
+    const checkbox = document.createElement('input');
+    checkbox.type = 'checkbox';
+    checkbox.classList.add('task-check');
+    checkbox.addEventListener('click', () => {
+      task.status(item, list);
+      todoList();
+    });
```
Comment (Gambit142) [inline modules/src/index.js:25]: - [ ] (Required) When I change the `Completed` status of a task and I reload the page, the status update is lost. Although the data is stored in the `localStorage,` there is no function or line of code to fetch data from the `localStorage`. I would like to advise you to always check if there is data in the localStorage upon reload of the webpage. If there is data stored in there, you can retrieve it and assign the `list` array to the data retrieved.  If there is none stored, you can set the `list` array to an empty array.

File: modules/src/index.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+document.querySelector('#task-Form').addEventListener('submit', (event) => {
+  event.preventDefault();
+  task.add(list);
+  todoList();
+});
+document.querySelector('.clear-btn').addEventListener('click', () => {
+  task.removeDone(list);
+  todoList();
+});
```
Comment (Gambit142) [inline modules/src/index.js:54]: - [ ] (Required) When I delete all  _**completed tasks**_ and I reload the page, the data is not retrieved from the `localStorage`. Although the data is stored in the `localStorage,` there is no function or line of code to fetch data from the `localStorage`. I would like to advise you to always check if there is data in the localStorage upon reload of the webpage. If there is data stored in there, you can retrieve it and assign the `list` array to the data retrieved.  If there is none stored, you can set the `list` array to an empty array.

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @rotshidzw 👋, 

Good job so far!

## Highlights
- All linters checks passed 💯 

However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- I noticed that when I tried to run `npm run start`, I got some errors. The error is a result of putting the `src` as a subfolder to the `modules` folder instead of the other way around. Remember that in your webpack.config.js file, you indicated entry as `'./src/index.js'` and not `'./modules/src/index.js'`. Kindly fix this error. 
 
- _Kindly check the comments under the review for other required changes._ 👍 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- You can also consider the optional suggestion I made under the review. 👍 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by codepantha — APPROVED** [review]
Hi @rotshidzw 👋,

## STATUS: APPROVED ✔️  

Well done implementing all the required changes 👍 👏 
Your project is complete now complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



### Highlights 🎯 
✔️ Descriptive PR
✔️ Meaningful commit messages
✔️ Linter checks

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AmaduKamara — COMMENTED** [review]
# Status: Approved 🟢🟢🟢 

Hi @rotshidzw,

YesYouAreGIF (4)

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! 🎉 

## Highlights

✅ Correct use of Gitflow
✅ No linter error
✅ Todo completed status change upon user interaction
✅ App design is cool 👍 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: dist/index.html [hunk 3]
```diff
@@ -1,38 +1,30 @@
...
-          <div class="to-do-list-section">
-              <h2>Today's To Do<i class="fas fa-sync icon1"></i></h2>
-          <div class="icons-1">
-              <input type="text" id="descr" class="input-1" placeholder="Add to your list">
-              <i class="fa fa-plus" aria-hidden="true"></i>
-          </div>
-              
-              
-              <ul class="todo-list">
-
```
Comment (ogagaoghene) [inline dist/index.html:29]: Everything is okay. Nice work. Your project is approved

**Review by ogagaoghene — COMMENTED** [review]
Nice work. Your work is complete.

**Review by ogagaoghene — COMMENTED** [review]
Your work is okay

**Review by ogagaoghene — COMMENTED** [review]
Nice work.Thumbs up.

**Pull request merged**